onnxruntime/core/providers/dml/DmlExecutionProvider/src/TensorDesc.cpp

```diff
@@ -212,12 +212,20 @@ gsl::span<const uint32_t> TensorDesc::GetStrides() const
 
 void TensorDesc::SetStrides(gsl::span<const uint32_t> strides)
 {
+    m_bufferTensorDesc.Strides = strides.empty() ? nullptr : strides.data();
```

Comment:
I might be blind, but why are we just assigning the raw pointer from a temporary input parameter instead of copying `strides` to `TensorDesc::m_strides`, and then setting `m_bufferTensorDesc.Strides` to `TensorDesc::m_strides`?

Reply:
:O I didn't even think about that, I just adapted the old code (which was doing `m_bufferTensorDesc.Strides = strides.data()`) to handle empty strides without even questioning if the old code was correct. But you're totally right, this code is not very safe. I suppose it works in practice because `m_bufferTensorDesc` isn't used outside of the scope of the enclosing function, but it's not ideal.